surveys/views.py

```diff
@@ -91,6 +93,62 @@ def post(self, request, **kwargs):
                       )
 
 
+class LocationList(ListView):
+    model = pldp_models.Location
+    template_name = "location_list.html"
+    context_object_name = 'locations'
+
+
+class LocationDetail(DetailView):
+    model = pldp_models.Location
+    template_name = "location_detail.html"
+    context_object_name = 'location'
+
+    def get_context_data(self, **kwargs):
+        context = super(LocationDetail, self).get_context_data(**kwargs)
+
+        location = context['location']
+
+        location_area = get_or_none(pldp_models.LocationArea, location=location)
+        location_line = get_or_none(pldp_models.LocationLine, location=location)
```

Comment:
Do we need this custom function (`surveys.utils.get_or_none`)? If we used `try`/`except`/`else` blocks for this conditional flow instead, we could avoid introducing a new module with a specialty function.

Reply:
I wrote this custom function to keep the code DRYer, as I'd be writing the same try/except block over and over. I've found this is an operation I've wanted in most Django projects, so it seems handy. Is there a reason I should write out those blocks repeatedly?

Reply:
Gotcha. I don't necessarily agree that the `try`/`except` flow violates DRY, since it's really only a couple lines longer than the `get_or_none` solution:

```python
location_area = get_or_none(pldp_models.LocationArea, location=location)

if location_area:
    ...
```

vs.

```python
try:
    location_area = pldp_models.LocationArea.objects.get(location=location)
except pldp_models.LocationArea.DoesNotExist:
    pass
else:
    ...
```

But in the end this is a matter of preference, so I'm fine going with the method if you prefer it. Agreed that I'm surprised there isn't a built-in way of retrieving `None` for missing objects using the `get` method in the ORM.